ONSdigital/eq-author-app
# Make questionnaire public or private

### What is the context of this PR?
<https://trello.com/c/Z2mYhATn/1150-make-questionnaires-public-private-6>
Add the concept of public and private questionnaires. Users can only see a private questionnaire if they are an owner or an editor. Questionnaire is public by default, and can be set to private in the sharing modal.

### How to review 
Check that everything works as expected and the tests make sense.

* Try opening a private questionnaire you dont have editor access on, you should be shown an unauthorized access page.
* Try deleting yourself as editor, you should be shown an alert.
